tests/pytests/unit/returners/test_local_cache.py

```diff
@@ -0,0 +1,16 @@
+from salt.returners.local_cache import _remove_job_dir
+from tests.support.mock import patch
+
+
+def test_remove_job_dir():
+    # Test that _remove_job_dir job will catch error
+    for e in (NotADirectoryError, OSError):
+        with patch("shutil.rmtree", side_effect=e("Node Corruption!")):
+            _remove_job_dir("cache.json")
```

Comment:
**suggestion** It's minor, but because it's explicitly called out as the desired behavior, I'd go ahead and ensure that log *is* called with the expected message.